ludwig/models/gbm.py

```diff
@@ -94,75 +98,106 @@ def forward(
-        #     encoder_output = encoder(input_values)
-        #     encoder_outputs[input_feature_name] = encoder_output
-
-        # concatenate inputs
-        inputs = torch.cat(list(inputs.values()), dim=1)
-
         # Invoke output features.
         output_logits = {}
         output_feature_name = self.output_features.keys()[0]
         output_feature = self.output_features[output_feature_name]
 
-        assert (
-            type(inputs) is torch.Tensor
-            and inputs.dtype == torch.float32
-            and inputs.ndim == 2
-            and inputs.shape[1] == len(self.input_features)
-        ), (
-            f"Expected inputs to be a 2D tensor of shape (batch_size, {len(self.input_features)}) of type float32, "
-            f"but got {inputs.shape} of type {inputs.dtype}"
-        )
-        # Predict using PyTorch module, so it is included when converting to TorchScript.
-        preds = self.compiled_model.model(inputs)
+        # If the model has not been compiled, predict using the LightGBM sklearn iterface. Otherwise, use torch with
+        # the Hummingbird compiled model. Notably, when compiling the model to torchscript, compiling with Hummingbird
+        # first should preserve the torch predictions code path.
+        if self.compiled_model is None:
+            # If `inputs` is a tuple, it should contain `(inputs, targets)``. When using the LGBM sklearn interface,
+            # `targets` is not needed, so it is discarded here.
+            if isinstance(inputs, tuple):
+                inputs, _ = inputs
+
+            assert list(inputs.keys()) == self.input_features.keys()
+
+            # The LGBM sklearn interface works with array-likes, so we place the inputs into a 2D numpy array.
+            in_array = np.stack(list(inputs.values()), axis=0).T
+
+            # Predict on the input batch and convert the predictions to torch tensors so that they are compatible with
+            # the existing metrics modules.
+            if output_feature.type() == NUMBER:
+                # Input: 2D eval_batch_size x n_features array
+                # Output: 1D eval_batch_size array
+                preds = torch.from_numpy(self.lgbm_model.predict(in_array))
+                logits = preds.view(-1)
+            else:
+                # Input: 2D eval_batch_size x n_features array
+                # Output: 2D eval_batch_size x n_classes array
+                probs = torch.from_numpy(self.lgbm_model.predict_proba(in_array))
+                if output_feature.type() == BINARY:
+                    probs = torch.logit(probs)[:, 1]
+                elif output_feature.num_classes == 2:
+                    probs = torch.logit(probs)
+                logits = probs
 
-        if output_feature.type() == NUMBER:
-            # regression
-            logits = preds.view(-1)
         else:
-            # classification
-            _, probs = preds
+            if isinstance(inputs, tuple):
```

Comment:
I can see us wanting to refactor this to use more interfaces and less if-else statements, but this is fine for now, I think.